### PR TITLE
Add AVIF Rewrite rules

### LINKED ISSUE #764: AVIF support: rewrite rules
### Scope a solution ✅ 
In `classes/avif/rewriterules/Apache`:
- Add new method for avif rules
- Use the new methods in the get method

In `classes/avif/rewriterules/nginx`:
- Add new method for avif rules
- Use the new methods in the get method

### Estimate the effort ✅
Effort XS

### PR DESCRIPTION
**Pull request by Miraeld**
## Description

This PR adds the rewrite rules management for the AVIF picture format

Fixes #764 

## Type of change

- New feature (non-breaking change which adds functionality).

## Is the solution different from the one proposed during the grooming?

No

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [x] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [ ] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

### CODE REVIEW

File: classes/Avif/RewriteRules/Apache.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Imagify\Avif\RewriteRules;
+
+defined( 'ABSPATH' ) || die( 'Cheatin’ uh?' );
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Apache.php]: You can remove this line, we don't need it in classes

File: classes/Avif/RewriteRules/Apache.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Add and remove rewrite rules to the .htaccess file to display AVIF images on the site.
+ *
+ * @author Gael Robin
+ */
+class Apache extends \Imagify\WriteFile\AbstractApacheDirConfFile {
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Apache.php]: You can import the class (`use`) before using it here

File: classes/Avif/RewriteRules/Apache.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<?php
+namespace Imagify\Avif\RewriteRules;
+
+defined( 'ABSPATH' ) || die( 'Cheatin’ uh?' );
+
+/**
+ * Add and remove rewrite rules to the .htaccess file to display AVIF images on the site.
+ *
+ * @author Gael Robin
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Apache.php]: You can remove the `@author` references, we don't use them anymore

**Review by remyperona — CHANGES_REQUESTED** [review]
You can do the same changes in the Nginx file

File: classes/Avif/RewriteRules/Apache.php (hunk 1 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Apache.php]: I've updated both classes :)

File: classes/Avif/RewriteRules/Apache.php (hunk 3 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Apache.php]: I've updated both classes :)

File: classes/Avif/RewriteRules/Apache.php (hunk 2 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Apache.php]: I've updated both classes :)

**Review by Khadreal — APPROVED** [review]
Looks good to me :)

File: classes/Avif/RewriteRules/Display.php [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+		global $is_apache, $is_nginx;
+
+		// Display AVIF?
+		$was_enabled = (bool) get_imagify_option( 'display_avif' );
+		// See \Imagify_Options->validate_values_on_update() for why we use 'convert_to_avif' here.
+		$is_enabled  = ! empty( $values['display_avif'] ) && ! empty( $values['convert_to_avif'] );
+
+		// Which method?
+		$old_value = get_imagify_option( 'display_avif_method' );
+		$new_value = ! empty( $values['display_avif_method'] ) ? $values['display_avif_method'] : '';
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		$old_value = get_imagify_option( 'display_next_gen_method' );
		$new_value = ! empty( $values['display_next_gen_method'] ) ? $values['display_next_gen_method'] : '';
```

File: classes/Avif/RewriteRules/Display.php [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+	/**
+	 * Add rules on plugin activation.
+	 */
+	public function activate() {
+		$conf = $this->get_server_conf();
+
+		if ( ! $conf ) {
+			return;
+		}
+		if ( ! get_imagify_option( 'display_avif' ) ) {
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		if ( ! get_imagify_option( 'display_next_gen' ) ) {
```

File: classes/Avif/RewriteRules/Display.php [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+			return;
+		}
+		if ( self::OPTION_VALUE !== get_imagify_option( 'display_avif_method' ) ) {
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		if ( self::OPTION_VALUE !== get_imagify_option( 'display_next_gen_method' ) ) {
```

File: classes/Avif/RewriteRules/Display.php [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+	/**
+	 * Remove rules on plugin deactivation.
+	 */
+	public function deactivate() {
+		$conf = $this->get_server_conf();
+
+		if ( ! $conf ) {
+			return;
+		}
+		if ( ! get_imagify_option( 'display_avif' ) ) {
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		if ( ! get_imagify_option( 'display_next_gen' ) ) {
```

File: classes/Avif/RewriteRules/Display.php [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+			return;
+		}
+		if ( self::OPTION_VALUE !== get_imagify_option( 'display_avif_method' ) ) {
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		if ( self::OPTION_VALUE !== get_imagify_option( 'display_next_gen_method' ) ) {
```

File: classes/Avif/RewriteRules/Display.php [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+	 * If display AVIF images via rewrite rules, add the rules to the .htaccess/etc file.
+	 *
+	 * @param  array $values The option values.
+	 * @return array
+	 */
+	public function maybe_add_rewrite_rules( $values ) {
+		global $is_apache, $is_nginx;
+
+		// Display AVIF?
+		$was_enabled = (bool) get_imagify_option( 'display_avif' );
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		$was_enabled = (bool) get_imagify_option( 'display_next_gen' );
```

File: classes/Avif/RewriteRules/Display.php [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+		// See \Imagify_Options->validate_values_on_update() for why we use 'convert_to_avif' here.
+		$is_enabled  = ! empty( $values['display_avif'] ) && ! empty( $values['convert_to_avif'] );
```
Comment (remyperona) [inline classes/Avif/RewriteRules/Display.php]:
```suggestion
		$is_enabled  = ! empty( $values['display_next_gen'] ) && ! empty( $values['convert_to_avif'] );
```

**Review by remyperona — CHANGES_REQUESTED** [review]
This PR is missing the instantiation and loading of the init method of the class, but I want to move to the container + event manager instead, so we could do it once they are in

File: classes/Avif/RewriteRules/Display.php (hunk 10 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 9 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 8 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 7 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 6 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 5 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

File: classes/Avif/RewriteRules/Display.php (hunk 4 above)
Comment (Miraeld) [reply classes/Avif/RewriteRules/Display.php]: It's been changed

**Pull request merged**